### ISSUE TITLE
Portmaster Offline Installer

### ISSUE THREAD

**Issue by FrostBlade5** [safing/portmaster#942]
I just installed windows in a VM, i could not install portmaster untill i enabled networking for the VM.
Problem is, after i enabled networking and installed portmaster i could see that windows has already made many connections to microsoft servers. This is a first privacy breach that SHOULD NOT happen.

Ofcourse portmaster downloads filterlists, etc, from the internet, BUT that does not mean that we can't have a portmaser offline installer that does prevent windows from phoning home upon installation, and ofcourse this offline installer would not be 100% "completed" it would download the filterlists as soon as it has network access, but really i HOPE that you can make this happen, offline installer would be really really good.

**Comment by Darthagnon** [comment]
Any news on an offline installer? I can't use offline installers in my network environment, and surely downloading resources like that means it could be prone to supply chain poisoning?

**Comment by firefoxlover** [comment]
This should be a high priority.

You could just somehow log what the installer is downloading and then make a script where all these files are moved from an archive to these directories.

Its not important that its up-to-date. When running it can check for updates for sure.

**Comment by redtrillix** [comment]
Yeah this would help tremendously, I live in Australia and for whatever reason I just can't download Portmaster using the online installer and i see it downloading at kbps in task manager, I've also tried in Arch on the AUR with the same results. 

Its not a  network bottleneck either cause I've got an internet connection of 100 mbps and my LAN being all Gigabit.

**Comment by firefoxlover** [comment]
For everyone having the same problem: I recommend Simplewall if you just need a Firewall for Windows. I dont know if it sufficiently blocks early-boot connections, maybe just use it before setting up Portmaster

**Comment by Rhea-3** [comment]
Simplewall has early-boot protection in options and is activated by default, which should block everything before BFE are loaded =) At least I didn't had any connections in a test VM.

Anyways, considering Portmaster: I wanted to give Portmaster a try today and I couldn't belive it really has no offline installer. This way another firewall is needed to install this firewall or the user doesn't care about privacy at all, which is somewhat ridiculus for a firewall advertising "privacy". Not very trustworthy in my opinion :confused:

**Comment by LoganDark** [comment]
You can use [Windows Firewall Control](https://www.binisoft.org/wfc) to lock down the built-in Windows Defender Firewall for a Portmaster install. This doesn't install any drivers or anything, it literally just acts as a front-end to the existing Windows Defender Firewall that will aggressively reject any attempts at external tampering. Then you can safely only allow Portmaster, and then properly lock down Portmaster, before disabling Windows Firewall completely.

Then remember to open regedit and inside `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Notifications\Settings\Windows.Defender.SecurityCenter` (create the key if it doesn't exist), set an `Enabled` DWORD to 0 to permanently disable those nagging notifications that "Windows Firewall is disabled, tap or click to turn it on". Literally nobody cares and literally nobody wants to turn it back on, lmao.

**Comment by iulko** [comment]
Any ETA on Offline installer?

**Comment by Raphty** [comment]
No ETA, this is something that will happen when we have the time for it.

**Comment by firefoxlover** [comment]
okay good to hear but why is this so complicated? The installer probably just downloads the source as an archive and moves the files, why cant this simply be replaced by a local archive? But no idea of windows installers, sorry

**Comment by iulko** [comment]
> okay good to hear but why is this so complicated? The installer probably just downloads the source as an archive and moves the files, why cant this simply be replaced by a local archive? But no idea of windows installers, sorry

#13
